### DISCUSSION TITLE
Board MKS_Tinybee v.1_0. Becomes inoperable! With additional features in Web Builder v0.9f. Be it SD or anything else!!

### DISCUSSION THREAD

**Discussion by LEOOZERAN** [grblHAL/core#438] [General]
Support for RS274 NGC expressions. Is it available? For boards (Devtronic_CNC_Controller(Blackpill 411) and MKS_tinybee_1_0.? The latter is NOT flashed! When adding any additional functions in Web Builder v0.9f Will all grblHAL functions finally be implemented ON the MKS_tinybee_1_0 board?. It can only work with standard settings !Otherwise it won’t work (several additions to Web Builder v0.9f) and the board becomes non-working. Please! Implement your additions on this board!!!

**Comment by terjeio** [comment]
> Support for RS274 NGC expressions. Is it available? For boards (Devtronic_CNC_Controller(Blackpill 411) and MKS_tinybee_1_0.?

Yes, all drivers supports this, except perhaps the STM32F103 with 128K flash.

> The latter is NOT flashed!

You enabled NGC expressions and it was not added? What is the `$I` output from that build?

> When adding any additional functions in Web Builder v0.9f Will all grblHAL functions finally be implemented ON the MKS_tinybee_1_0 board?

All is impossible since the ESP32 has limited number of pins available. I have just added _Keypad_ plugin support, some other options may become available if specific feedback is given and there is pins or other relevant resources available. 3rd party plugins support might not happen due to the ESP build system causing problems for me.

> It can only work with standard settings

I fixed a regression yesterday that caused the firmware to hang if the WebUI plugin was enabled. If you still experience that please provide more detailed feedback, I simply do not have time to test every combination myself.

**Reply by LEOOZERAN** [reply]
I'm planning to create my own clone, MKS_tinybee_1_0 boards. With minor changes. And it will have its own standalone controller via RX and TX. QUESTION! I am interested in the most important thing. Macro support. And reading from SD card. Nothing else! In the standard version. Are these features available??? The board will be created specifically for GRBL HALL! Thanks for the answer!

**Reply by terjeio** [reply]
> Are these features available??? 

Yes.

> The board will be created specifically for GRBL HALL! 

If you plan to publish the design then create a new map file and submit a PR for it. There is no need to use any of the current maps.

**Reply by LEOOZERAN** [reply]
My board will be entirely based on the .MKS TinyBee board (card). Just add buffers, optocouplers, etc. (in the form of outputs and inputs. Board, 6-axis machine (FluidNC). I don’t like it. But the MKS is very attractive for processing. My independent controller, it is planned to work both on the MKS and on the Devtronic CNC controller. Moreover, the control will be both milling, and transition to turning. I have a project. Automatic holder for turning tools. (revolver).It will be controlled by a stepper motor. For example, axis B. This is why I need macros. When the project is partially ready, I plan to post it. For everyone to see! On your site! Thank you for your work!  Hello from Ukraine
